src/components/ui4/BlogCards.tsx

```diff
@@ -0,0 +1,44 @@
+import { BlogPost, BlogPostGeo } from 'edge-info-server'
+import React from 'react'
+import { ListRenderItem } from 'react-native'
+import { useSafeAreaFrame } from 'react-native-safe-area-context'
+
+import { useHandler } from '../../hooks/useHandler'
+import { infoServerData } from '../../util/network'
+import { useTheme } from '../services/ThemeContext'
+import { BlogCard } from './BlogCard'
+import { CarouselUi4 } from './CarouselUi4'
+
+export interface Props {
+  countryCode?: string
+}
+
+export const BlogCards = (props: Props) => {
+  const { countryCode } = props
+  const { width } = useSafeAreaFrame()
+  const theme = useTheme()
+
+  const [blogCards, setBlogCards] = React.useState<BlogPost[]>([])
+
+  const renderBlog: ListRenderItem<BlogPost> = useHandler(({ item }) => <BlogCard blogPost={item} />)
+
+  // Check for BlogCards from info server:
+  React.useEffect(() => {
+    if (countryCode == null) return
+
+    const filteredBlogPostsGeo = (infoServerData.rollup?.blogPostsGeo ?? []).filter((blog: BlogPostGeo) => {
+      const { countryCodes: includeCountryCodes = [], excludeCountryCodes = [] } = blog
+
+      const isCountryInclude =
+        includeCountryCodes.length === 0 || (includeCountryCodes as string[]).map(includeCountryCode => includeCountryCode.toUpperCase()).includes(countryCode)
+      const isCountryExclude =
+        excludeCountryCodes.length > 0 && (excludeCountryCodes as string[]).map(countryCode => countryCode.toUpperCase()).includes(countryCode)
```

Comment:
use `.some()` instead of `.map` and `includes()`. More efficient, clean, and easier to read.